### ISSUE TITLE
Support the DNScrypt protocol

### ISSUE THREAD

**Issue by tarcieri** [bluejekyll/trust-dns#4]
It would be nice if this server supported the DNScrypt protocol:

https://dnscrypt.org/

cc @jedisct1

**Comment by bluejekyll** [comment]
I'm definitely planning on supporting DNSSEC, I can probably do this too.

**Comment by tarcieri** [comment]
Awesome. The use case is a bit orthogonal to DNSSEC (confidentiality and integrity within an "intranet"-style deployment between clients and a DNS server), but I would certainly consider it a welcome addition given your present roadmap.

**Comment by bluejekyll** [comment]
Yeah, sorry, I get that it serves a different purpose, but it lines up with the set of features I want in the platform. I only get to work on this in my spare time at the moment, so it might be a while before I get all that in here.

**Comment by bluejekyll** [comment]
FYI, in issue #38 there is a suggesting of first supporting https://tools.ietf.org/html/rfc7858.

Would people want to weigh in on that? I'll probably only end up really supporting one for a while (as either will fit my needs).

**Comment by bluejekyll** [comment]
btw, the dnscrypt branch has some initial certificate code. so this is maybe 25% done at this point.

**Comment by divergentdave** [comment]
Encryption and authentication of stub-to-recursive traffic is commonly done with DoH now, which we support already. I suggest we close this issue as not planned.

**Comment by alterstep** [comment]
Client IP hiding -> anonymized DNSCrypt is the only good option with many relays.

And DNSCrypt is at version 2026 (PQDNSCrypt, post-quantum) now.

**Comment by djc** [comment]
> Encryption and authentication of stub-to-recursive traffic is commonly done with DoH now, which we support already. I suggest we close this issue as not planned.

Seems there are still people who want it?

I do not plan on either working on or reviewing it in the near future, but we can keep this open for coordination between people who do want it.
